src/System.Collections.Specialized/src/System/Collections/Specialized/NameValueCollection.cs

```diff
@@ -111,22 +112,22 @@ protected void InvalidateCachedArrays()
             _allKeys = null;
         }
 
-        private static string GetAsOneString(ArrayList list)
+        private static string? GetAsOneString(ArrayList? list)
         {
             int n = (list != null) ? list.Count : 0;
 
             if (n == 1)
             {
-                return (string)list[0];
+                return (string?)list![0];
```

Comment:
I assumed bang is appropriate than Debug.Assert() in this case and similar cases below

Reply:
Well in this case, since you're sure that list will not be null in this if block and in the `else if (n > 1)` block, you can safely use `Debug.Assert(list != null`). The reason for that is because the only way of getting an n with value >= 1 is if `list != null`.

Reply:
Well yes, my point is it is very obvious that it will not be null, so doesn't make sense to assert